Change Units on Noise Floor

arrmo opened the pull request:
I may be wrong, but I think this should just be dBm (i.e. integrated over the carrier bandwidth). The numbers look much too high to me to be on a per Hz basis.

If I am correct here, this should also be updated in the web interface.

Thanks!

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

murrant commented:
Which sensor are you talking about?  There are many noise floor sensors, some in dbm, some in Hz.

SourceDoctor commented:
@armo @murrant  dBm/Hz can be correct ...
especially for a "noise floor"
if value is high it's a good indicator this value is in dBHz.
dB/Hz i have never seen in my praxis. What says the data sheet?
For for generric Noisefloor i only know dBHz not dB/Hz, so i think it should be changed to dBHz

short excursion:
dBHz - MHz = dB
which means:
spectral noisefloor - used Bandwidth = Carrior to Noise ratio
Example:
70 dBHz - 60 (MHz logarithmic) = 10dB C/N


as you see the dBHz value has to be high

arrmo commented:
Sorry, not quite following your comment - but may be me. Let me try to explain my thought though.

If this is really noise floor ... well, then starting from the thermal noise floor (-174 dBm/Hz), and if we assume a NF (noise figure) of ~ 10 dB (guess here a bit), then the noise floor would be roughly -164 dBm/Hz. Integrating over a 20 MHz carrier BW, this adds 73 dB for 10*log(BW) ... puts the integrated noise floor at roughly -91 dBm (not per Hz, this is integrated over BW). As the results are pretty much in this range, it seems to me that this measurement really is integrated ... so dBm, not per Hz.

BTW, also I do see from iwinfo on the router, noise reported as (example) -89 dBm.

Thoughts?

Thanks!

SourceDoctor commented:
your assume sounds right on reading over it.
-91dBm . -89dBm are not such far away ... especially for classical noise floor in RF equipment.
Keep in mind noise floor depends on many many many parameters, also on component temperatures, age of components, quality, RF injections into components and antenna, quality of isolation of RF parts, and so on... so a noise floor up to nearly -85dBm is quiet a normal dimension

arrmo commented:
Completely agree! And the 10 dB NF was a guess on my part ;-).

Also of course, this measurement is not just NF - it also detects environment noise floor (which is good).

SourceDoctor commented:
problem with the unit here and it's usage is ...
is it the real noise floor? so it's dBm. 
is it the noise figure of the system? so it's dBHz

would be useful to recheck where this unit is used here and recheck the datasheets of the devices.
Which potentially means correction of unit description on several devices and potential add another unit

arrmo commented:
Ahh, got your point now 😆.

Yes, agreed - it depends on what the / each device reports ... noise figure, floor, spectral density, etc. Hmmm ... not sure how to confirm that, any thoughts? At least for the devices I'm playing with, they are reporting integrated noise floor, but as you note - this may or may not be consistent.

Thanks!

SourceDoctor commented:
i appreciate your PR ... but i think there is much more to do as first thought ;-)
especially grepping through existing devices to see which of them use this unit
and looking into their datasheets if it is used correctly.
Adding new units is no problem, but it has to be used correctly, otherwise it makes no sense ...
Define there wrong units is like define Disc capacity in Mb/s or worse MB/s :D

arrmo commented:
Yes, agreed! And no worries if the PR needs to be rejected, I just noticed that this is "wrong" ... but for my devices. To your point, it doesn't mean that this is the case generically - I wasn't thinking that widely 😆.

Thanks!

SourceDoctor commented:
it's good to fix errors :) we appreciate this. if you've found a missing unit for this on to your device. Feel free to fix it.

arrmo commented:
Well, my devices (2 OS's) report dBm ... I guess what I don't know is what other devices are reporting - meaning, what is the generic answer? I'm OK with dBm, or dBm/Hz, if one or the other is used more widely. Do you happen to know? I admit, not sure what other devices are reporting 😞.

Thanks!

SourceDoctor commented:
i took a short look over code. There are several devices which use this unit. i think this has to be overlooked.

arrmo commented:
NP, I'm good with whatever you conclude - just noticed it, thought I'd try to correct it.

BTW, finding a few other "odd" unit related items (like frequency in kMHz) - these are minor of course, just trying to help clean up if I can. That said, I REALLY like the Wireless Sensors approach. Just took a bit of time to figure it out, some IBM issues ... 😆.

*IBM = Idiot Behind the Machine => me!

SourceDoctor commented:
feel free to correct everything you find 
on some things it may more to do than thought on first look, but it will be appreciated :)

for direct questions we'll help you in community.librenms.org or in discord 

to come back to this PR it would be nice if you would fix the units all over

arrmo commented:
Sure, will try to correct these, NP at all! I admit, I tried to run the maze of finding where the units are defined in the code .. felt like a mouse after cheese. LMAO!

Will keep hunting, NP - if you have any pointers where to look, it would be greatly appreciated.

Thanks!

SourceDoctor commented:
i found them with grep ;-)

arrmo commented:
Dang it! I tried that - hmmm, that's likely me. Will dig!

arrmo commented:
OK, easy fix for the GUI display ... do you want me to go ahead and submit a PR (change it to dBm?). FYI, still need to find the RRD generation - that's different.

And frequency - that's different also, that's about converting > 1000 (or 1024) to "k".

Thanks!

SourceDoctor commented:
fix what you find :)

murrant commented:
The graphs: rrdtool does automatic prefixing... and sucks at some times.  You could maybe convert the data to Hz in the graph and hopefully the graph would prefix correctly.  Or probably someway to disable the graph prefixing.

arrmo commented:
Actually, frequency is fine in rrd, it's just the dBm/Hz I need to fix there, will find it.

Frequency is messed up in the web GUI, Wireless > Frequency. Will hunt for that too.

Attach the updates here (to this PR), so all in one place?

Thanks!

arrmo commented:
OK, added some updates - found where (and corrected) the display units for noise floor. Also, updated the formatting of wireless frequency, so it doesn't try to format_si GHz to kMHz.

But ... I'm not 100% happy with the second one. I added a check similar to one that is already there. That part is fine, but ... I don't like checking the sensor_type => really, it should be for all wireless sensors, agreed? But I can't figure out how to check that. And I can't just check for a sensor_type of frequency, as then it will hit other items (like Raspberry Pi clock frequency). Thoughts?

Thanks!

PS: Yep, still to fix the units in the noise-floor rrd plots 😄.

arrmo commented:
OK, found where I *think* the units are set for rrd 😄.
`includes/html/graphs/device/wireless_noise-floor.inc.php
`

Tried to update the units, but I admit - not sure how to force re-generation of the rrd plots.

arrmo commented:
Never mind! 😆. It's in 2 places, not quite sure why, but fixed both, updated.
```
includes/html/graphs/device/wireless_noise-floor.inc.php
includes/html/graphs/wireless/noise-floor.inc.php
```

I think this is pretty clean now, just not 100% happy with the sensor check (that I noted above). Open to suggestions on a better check.

Thanks!

murrant commented:
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

murrant commented:
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1
